### PR TITLE
Fix test.

### PR DESCRIPTION
**Pull request by ghislainfourny**
Fixing the test. The dynamic context should automatically cascade the count from the RDD.

### CODE REVIEW

File: src/main/java/sparksoniq/semantics/DynamicContext.java [hunk 1]
```diff
@@ -202,6 +202,9 @@ public Item getLocalVariableCount(String varName) {
         if (_localVariableCounts.containsKey(varName)) {
```
Comment (ghislainfourny) [inline src/main/java/sparksoniq/semantics/DynamicContext.java]: The name of this function could simply by getVariableCount(). It just returns the count irregardless of whether the variable is local, RDD or DF.

Comment (CanBerker) [reply src/main/java/sparksoniq/semantics/DynamicContext.java]: Very nice ! I like the name and the functionality

File: src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/sequences/aggregate/CountFunctionIterator.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public Item next() {
                 if (iterator instanceof VariableReferenceIterator) {
                     VariableReferenceIterator expr = (VariableReferenceIterator) iterator;
                     this._hasNext = false;
-                    return _currentDynamicContext.getLocalVariableCount(expr.getVariableName());
+                    return _currentDynamicContext.getVariableCount(expr.getVariableName());
                 }
```
Comment (CanBerker) [inline src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/sequences/aggregate/CountFunctionIterator.java]: I feel that this entire if block with the instanceof check can be taken out of the if(!iterator.isRDD()) block. The current implementation with getVariableCount materializes through count() or size() in any scenerio, making the isRDD check redundant for the execution of VariableReferenceIterator within CountFunctionIterator.

Comment (ghislainfourny) [reply src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/sequences/aggregate/CountFunctionIterator.java]: This is a very good point. I have reorganized the cascade accordingly.

I also removed a few elses, which are unnecessary when the blocks return from the method.

**Comment by ghislainfourny** [comment]
@CanBerker feel free to merge this into your branch, and then we can proceed with merging your branch into the master.

**Pull request merged**